### PR TITLE
Check for Mach-O using byte strings to allow case of non unicode chars

### LINKED ISSUE #529: setRelativeReferencePaths() Error when trying to freeze wxPython app
Hi,

OS - MacOS Mojave
Python version: 3.7.5

When trying to freeze a wxPython application cx_Freeze fails with following traceback

`
File "../../Python.framework/Versions/3.7/lib/python3.7/site-packages/cx_Freeze/macdist.py", line 180, in setRelativeReferencePaths
if "Mach-O" in p.stdout.readline().decode():
UnicodeDecodeError: 'utf-8' codec can't decode bytes in position 281-282: invalid continuation byte
`

Any suggestions on how to proceed?

### PR DESCRIPTION
**Pull request by QuantumEntangledAndy**
The recent change of not encoding byte strings in bdist_mac is breaking the build. Methods using bytes should not be mixed with str. This patch unifies the relevant code to use all bytes.

NOTE: There is one step where I choose to `decode`

```python
else:
                        files.append(name.decode())
```

This is because 
```python 
filePath = os.path.join(self.binDir, fileName)
```

The `fileName` is generated from type `str` (from `os.walk`) but when we add extra files in the loop they were added as type bytes.

Currently I decode back to string then append to the list, but alternatively it may be better to encode the results of `os.walk` so that it is all bytes throughout.

### CODE REVIEW

**Comment by QuantumEntangledAndy** [comment]
I have a further patch that gets `@loader_path` working (see #634) BUT when I came to merge I had conflicts from your changes to not decode. When I pulled your changes I ran into multiple errors related to using `str` when `bytes` was expected.

If you want I can roll the addition of `@loader_path` into this pull request too. Or I can wait until this is merged and pull request of that.

**Comment by marcelotduarte** [comment]
That is my patch. It was untested and I forget to declare this. In the issue #529 I requested tests but no one answered. As you use mac, if you find it better to revert the patch and you fix the issue, I even prefer it.

**Comment by QuantumEntangledAndy** [comment]
I actually had #529 and had a different fix. Rather than not decoding I did this instead
```
try:
    line = p.stdout.readline().decode()
except UnicodeDecodeError:
    continue
if "Mach-O" in line:
```

Here I assume that if we can't decode it is not Mach-O. Which fixes #529 in my case I am not sure what the original case was.

We have two choice

1. resolve by changing all to bytes (many changes, but possibly more flexible)

2. resolve by skipping any files where it is not decodable (less changes, but possibly not as flexible).

Which would you prefer?

**Comment by marcelotduarte** [comment]
I reverted my patch. Please send your patch, your approach is ok to me, the option 2.

**Comment by QuantumEntangledAndy** [comment]
I have made this pull request work as indicated in option 2.

Fixes #529

**Comment by QuantumEntangledAndy** [comment]
Actually I am thinking of an option 3.

Use str everywhere except when testing for Mach-O this will let us test for Mach-O in files with non unicode bytes, while still not requiring many changes

**Comment by marcelotduarte** [comment]
ok, put it in another PR

**Comment by QuantumEntangledAndy** [comment]
A completely separate PR? Or ammend this one?

**Comment by marcelotduarte** [comment]
Can be in this PR, but change the title to not confuse anthony.

**Comment by QuantumEntangledAndy** [comment]
Latest commit is using option number 3. This allows searching for Mach-O even if there are non unicode chars but keeps the rest of the code as str

**Review by marcelotduarte — APPROVED** [review]

**Pull request merged**
